crates/wasi-common/src/hostcalls_impl/fs.rs

```diff
@@ -162,7 +164,7 @@ pub(crate) unsafe fn fd_read(
         .as_descriptor_mut(wasi::__WASI_RIGHTS_FD_READ, 0)?
     {
         Descriptor::OsHandle(file) => file.read_vectored(&mut iovs),
-        Descriptor::Stdin => io::stdin().lock().read_vectored(&mut iovs),
+        Descriptor::Stdin => io::stdin().read_vectored(&mut iovs),
```

Comment:
Is there any reason why we're dropping the `lock` here?

Reply:
Oh, this was just an incidental code cleanup; `read_vectored` [does a lock automatically](https://github.com/rust-lang/rust/blob/master/src/libstd/io/stdio.rs#L338).